src/twisted/python/test/test_components.py

```diff
@@ -587,8 +587,11 @@ def boo():
         """
 
 
+
 @implementer(IProxiedInterface)
-class Yayable(object):
+class Yayable(object):  # type: ignore[misc]
+    # class does not implement Attribute ifaceAttribute
+    # so we need to turn off mypy warning
```

Comment:
```suggestion
class Yayable(object):
   # This is replaced by each test. Here we add a default value. 
    ifaceAttribute = object()
```

Reply:
@adiroiban I made the change you suggested, but then the test failed with:

```
[FAIL]
Traceback (most recent call last):
  File "/Users/crodrigues/Twisted/src/twisted/python/test/test_components.py", line 907, in test_attributeCustomization
    self.assertFalse(hasattr(yayable, 'ifaceAttribute'))
  File "/Users/crodrigues/Twisted/src/twisted/trial/_synctest.py", line 390, in assertFalse
    super(_Assertions, self).assertFalse(condition, msg)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/case.py", line 699, in assertFalse
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: True is not false
```

So the test explicitly checks that ifaceAttribute is not there, before adding that attribute.

I'm not sure if it is OK to change the test?

Reply:
I think that we can change the tests as the 'hasAttr' check is not 100% correct.

As we have seen here, even if the instance member is removed, hasAttr will still  return True...as the class has the attrribute.

I have proposed a patch to improve this test and make it explicit that only the instance attribute is removed.